sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -848,4 +851,19 @@ object DDLUtils {
       }
     }
   }
+
+  private[sql] def checkFieldNames(table: CatalogTable): Unit = {
+    val serde = table.storage.serde
+    if (serde == HiveSerDe.sourceToSerDe("orc").get.serde) {
+      OrcFileFormat.checkFieldNames(table.dataSchema)
+    } else if (serde == HiveSerDe.sourceToSerDe("parquet").get.serde) {
```

Comment:
We could have different Parquet serde. For example, `parquet.hive.serde.ParquetHiveSerDe` and `org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe`. How about ORC?

Reply:
AFAIK, it's only `org.apache.hadoop.hive.ql.io.orc.OrcSerde`.  I checked again whether Apache ORC 1.4.0 have some renamed one under `hive-storage` API or not. But, it doesn't have it.

For parquet, I'll handle that too.